packages/vite/src/node/ssr/runtime/__tests__/fixtures/worker.invoke.mjs

```diff
@@ -0,0 +1,57 @@
+// @ts-check
+
+import { BroadcastChannel, parentPort } from 'node:worker_threads'
+import { fileURLToPath } from 'node:url'
+import { ESModulesEvaluator, ModuleRunner } from 'vite/module-runner'
+
+if (!parentPort) {
+  throw new Error('File "worker.js" must be run in a worker thread')
+}
+
+/** @type {import('vite/module-runner').ModuleRunnerTransport} */
+const transport = {
+  async invoke(/** @type {import('vite').HotPayload} */ event) {
+    const hotPayloadData = event['data']
+
```

Comment:
Could you change this to send messages and receive the response to / from the parentPort by using [birpc](https://github.com/antfu/birpc) or something else here? To make the test not to fail even if we change the internals of the module runner.

Reply:
Done, please have another look 🙂🙏